lib/base/configtype.hpp

```diff
@@ -48,10 +49,10 @@ for (const auto& object : objects) {
 	int GetObjectCount() const;
 
 private:
-	typedef std::map<String, intrusive_ptr<ConfigObject> > ObjectMap;
+	typedef std::unordered_map<String, intrusive_ptr<ConfigObject> > ObjectMap;
 	typedef std::vector<intrusive_ptr<ConfigObject> > ObjectVector;
 
-	mutable boost::mutex m_Mutex;
+	mutable boost::shared_mutex m_Mutex;
```

Comment:
Please use `std::timed_shared_mutex` instead of `boost::shared_mutex`. `boost::shared_mutex` uses an exclusive mutex and condition variables to emulate a shared mutex, while `std::shared_timed_mutex` can use the proper pthread rwlock primitives which were significantly faster in my tests for #9627.